torchao/csrc/cuda/sparse_marlin/mem.h

```diff
@@ -19,6 +19,28 @@
 #include "base.h"
 
 namespace torchao {
+
+#ifdef USE_ROCM
```

Comment:
Should we gate on a specific ROCm version like we do for CUDA?

Reply:
Good point! What we need is a GPU arch check instead of rocm version check. I have added a GPU architecture check in the setup.py file. As a result, the kernel will now only be built for the MI300X architecture.